### ISSUE TITLE
Kondo doesn't report unused namespaces that don't have aliases

### ISSUE THREAD

**Issue by camsaul** [clj-kondo/clj-kondo#2413]
This is with Kondo `2024.09.07`:

```clj
(ns x
  (:require
   [clojure.test]))

(defn whatever
  "Whatever."
  [])
```

```sh
$ clj-kondo --lint src/x.clj
linting took 158ms, errors: 0, warnings: 0
```

doesn't warn about `clojure.test` being unused, even tho it's definitely unused. Not sure if this is intentional or not but I would have thought it would trigger a warning

**Comment by NoahTheDuke** [comment]
I think it doesn't because loading namespaces can have side effects (multi methods, protocols, setting vars, etc).

**Comment by borkdude** [comment]
This is intentional: not having an alias or refer is interpreted by clj-kondo as: this namespace is loaded for side-effect, so @NoahTheDuke is correct.
